code/modules/unit_tests/crayons.dm

```diff
@@ -0,0 +1,12 @@
+/// Makes sure that crayons have the correct names based on their flavour.
+
+/datum/unit_test/crayon_naming/Run()
+	var/list/bad_crayons_list = list()
+	for(var/i in typesof(/obj/item/toy/crayon))
+		var/obj/item/toy/crayon/this_crayon = i
+		if(!findtext(initial(this_crayon.name),this_crayon.crayon_color))
+			bad_crayons_list += "[this_crayon.type]"
+		if(!L.len)
+			return //passed!
+		Fail("The following crayons do not have their crayon_color in their initial name: [L.Join(",")])
```

Comment:
Don't do this. Just call `Fail` whenever you get to an invalid name. It will continue the test.